### PR TITLE
Feature/js19 259 login/register

### PR DESCRIPTION
**Pull request by karbik-creator**

### CODE REVIEW

**Review by elena-dragunova — APPROVED** [review]

**Review by panov-aleksey-saritasa — APPROVED** [review]
Please, take a look at issues from SonarCloud

**Review by timofei-maximov-saritasa — APPROVED** [review]

**Review by golovachev-yaroslav — APPROVED** [review]

File: vanilla/src/index.css [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+.header__wrapper {
+  display: flex;
+  justify-content: right;
+}
+
+.userName {
+  margin-right: 20px;
+
+  color: var(--main-color);
+  font-size: 22px;
```
Comment (danilkukryakov) [inline vanilla/src/index.css:20]: We prefer to use relative units for font size (`rem` in this case will be fine).
Check our [wiki](https://frontendwiki.saritasa.rocks/pages/development/languages/css.html#use-relative-units).

File: vanilla/src/index.css [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+/* Form styles */
+
+.forms {
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
```
Comment (danilkukryakov) [inline vanilla/src/index.css:29]: Is it possible to use another approach for elements positioning? Try to use flex-box instead ('display: flex' + 'justify-content: center' and `align-items: center`.
Usually absolute positioning shouldn't be used for page layout grid. 
A better use case for `position: absolute` is for small elements positioning (e.g. floating action button).

File: vanilla/src/index.css [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  display: block;
+}
+
+.formAuth {
+  display: none;
+
+  min-height: 370px;
+  padding: 30px;
+
+  text-align: end;
```
Comment (danilkukryakov) [inline vanilla/src/index.css:42]: Maybe use `text-align: right` instead.

File: vanilla/src/index.css [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  border-radius: 15px;
```
Comment (danilkukryakov) [inline vanilla/src/index.css:44]: If we have the same border radius in our app (as we usually do) we can move its value to CSS variable (the same way you do with `--main-color`).

File: vanilla/src/index.css [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  height: 100%;
+
+  background-color: var(--hide-color);
+}
+
+.backgroundDrop.open {
+  display: block;
+}
+
+/* Buttons styles */
```
Comment (danilkukryakov) [inline vanilla/src/index.css:79]: As I can see you have several block (or groups) in this `.css` file. Maybe we should split them in separate files.
However, I don't think it's a big concern now as you have relatively small file. Anyway, don't forget to do it later if necessary.

File: vanilla/src/index.html [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  <title>App</title>
+  <script type="module" defer src="./index.ts"></script>
+</head>
+
+<body>
+  <header class="header">
+    <div class="card-panel teal lighten-2 ">
+      <div class="container header__wrapper">
+        <div class="userName"></div>
+        <button type="button" class="waves-effect waves-light btn loginBtn" data-type="open-sign-in">
```
Comment (danilkukryakov) [inline vanilla/src/index.html:20]: Please, don't mix three naming approaches for CSS classes. Chose one and stick to it.
According to our [wiki](https://frontendwiki.saritasa.rocks/pages/development/languages/css.html#bem) we prefer BEM naming convention. So I recommend you to use it in your project.

Comment (danilkukryakov) [reply vanilla/src/index.html:20]: I know some CSS classes are from library. My comment was more about `.loginBtn` and other classes written in `camelCase`.

File: vanilla/src/index.html [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  <div class="forms">
+    <form class="col s12 formAuth formAuth__signIn">
+      <button type="button" class="btn-flat" data-type="open-sign-up">
+        sign up
+        <em class="material-icons right" data-type="open-sign-up"> east </em>
+      </button>
+      <div class="row">
+        <div class="input-field col s12">
+          <input id="email_inline" type="email" class="validate" name="email" />
+          <label for="email_inline">Email</label>
```
Comment (danilkukryakov) [inline vanilla/src/index.html:39]: Usually we should link `input` and `label` elements. So when user click on label element the input element is focused.
There are two approaches of doing this. You can find examples [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#examples).
I recommend you to use the first one (when label element wraps input).

**Review by danilkukryakov — APPROVED** [review]

**Review by gevorg-gevorgyan-saritasa — APPROVED** [review]

**Pull request merged**